debezium-core/src/main/java/io/debezium/transforms/filter/Engine.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms.filter;
+
+import org.apache.kafka.connect.connector.ConnectRecord;
+
+/**
+ * The interface serves as an abstraction of expression language engine.
+ *
+ * @author Jiri Pechanec
+ */
+public interface Engine {
```

Comment:
This reminds me a little bit of JSR 223. How about making this our primary entry point? That'd have the advantage that people could use any compatible scripting language.

Reply:
Ok, so I don't think we'd get the auto-magic resolving of "properties" of records here via 223. Or rather, it might depend on the chosen impl. Can you please try out GraalVM JS, it has a JSR 223 binding. This might become a popular choice. I'd like to see how this would work. Would have have to resolve those props ourselves? Might also be there's a resolver contract of sorts for which we could plug in an impl based on records.

Reply:
Let me explain - I definitely intend to add support for JSR 223 in future. Right now I wanted to roll out a first implementation of scripting with the full power of Groovy, that's why I opted for native API. Even as it seems now to not be really important I guess we will need to slightly extend the DSL to handle schemas nicely.

So I prpose first to get this out and then create a follow up Jira for JS support. I understand why it is useful but I'd like first to check the API thoroughly as we might need to provide more elaborate bindings implementation that would allow the same syntax as we have with groovy right now. I need to now if JS engine has its own API that would allow us to inject the resolver or whether we would need to write Bindings in that way that they would return lmabdas/vs target values.

Reply:
So do you consider `Engine` a user-facing contract? What would speak against coding against JSR 223 but use Groovy as the sole supported impl at this point? I'd like to avoid cornering ourselves by supporting just one thing and then having to change user-facing contracts later on.

>  I need to now if JS engine has its own API...

Yes, of course. But that'd be a very valuable evaluation to do, and it could be done right now.